connector/connect/src/main/scala/org/apache/spark/sql/connect/dsl/package.scala

```diff
@@ -64,6 +64,36 @@ package object dsl {
           .build()
     }
 
+    /**
+     * Create an unresolved function from name parts.
+     *
+     * @param nameParts
+     * @param args
+     * @return Expression wrapping the unresolved function.
+     */
+    def fun(nameParts: Seq[String], args: Seq[proto.Expression]): proto.Expression = {
```

Comment:
I would name it `func` as that's (much) more common in the codebase as far as I can tell.

Reply:
this is just in the DSL package that is used for testing. I'll check what Catalyst is doing.

Reply:
the catalyst DSL has a similar `callFunction`

Reply:
For a bit of more context, one thing is that we use `snake_case` to match w/ SQL function names (see `Column` or `functions.scala`). This kind of naming rule is already mixed in our existing SQL DSL (see also `org.apache.spark.sql.catalyst.package`). Should probably pick one and stick to that.

In the past, we followed `camelCase` in both DSL, `Column` and `functions.scala`. After that, we renamed them all to `snake_case` for SQL compatibility in `Column` and `functions.scala` (so the new DSL added follows  `snake_case` at `org.apache.spark.sql.catalyst.package`)

Therefore, I tend to use `snake_case` in this DSL case too but I am fine if others (or you) feel strongly this is better.

Reply:
maybe name the two methods `function`?
 https://github.com/apache/spark/blob/ff7ab34a6957965d74504ed1d36de21d1ad7319c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala#L368-L369

don't feel strong about the naming

Reply:
Oh nice, I definitely missed renaming the other overall `callFunction` as well. TBH I'm not sure what the right approach is here because the catalyst DSL calls it `callFunction` instead of `call_function` 🤷

Reply:
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala#L250-L256 Please let me know what you want to do.